core/src/main/scala/kafka/server/BrokerFeatures.scala

```diff
@@ -0,0 +1,187 @@
+ * of certain versions. After this happens, any external client that queries the Broker to learn the
+ * feature versions will at some point start to see the new value for the finalized minVersionLevel
+ * for the feature. The external clients are expected to stop using the deprecated versions at least
+ * by the time that they learn about it.
+ *
+ * Here is how the above code change needs to be done:
+ * In order to deprecate feature version levels, in the supportedFeatures map you need to supply a
+ * specific firstActiveVersion value that's higher than the minVersion for the feature. The
+ * value for firstActiveVersion should be 1 beyond the highest version that you intend to deprecate
+ * for that feature. Whenever the controller is elected or the features are finalized via the
+ * ApiKeys.UPDATE_FEATURES api, the feature version levels in the closed range:
+ * [minVersion, firstActiveVersion - 1] are automatically deprecated in ZK by the controller logic.
+ *
+ * Example:
+ * - Let us assume the existing finalized feature in ZK:
+ *   {
+ *      "feature" -> FinalizedVersionRange(minVersionLevel=1, maxVersionLevel=5)
+ *   }
+ *   Now, supposing you would like to deprecate feature version levels: [1, 2].
+ *   Then, in the supportedFeatures map you should supply the following:
+ *   supportedFeatures = {
+ *     "feature" -> SupportedVersionRange(minVersion=1, firstActiveVersion=3, maxVersion=5)
+ *   }
+ * - If you do NOT want to deprecate a version level for a feature, then, in the supportedFeatures
+ *   map you should supply the firstActiveVersion to be the same as the minVersion supplied for that
+ *   feature.
+ *   Example:
+ *   supportedFeatures = {
+ *     "feature" -> SupportedVersionRange(minVersion=1, firstActiveVersion=1, maxVersion=5)
+ *   }
+ *   The above indicates no intent to deprecate any version levels for the feature.
+ *
+ * STEP 2:
+ * =======
+ *
+ * After the first step is over, you may (at some point) want to permanently remove the code/logic
+ * for the functionality offered by the deprecated feature versions. This is the second step. Here a
+ * subsequent major Kafka release is made with another Broker code change that removes the code for
+ * the functionality offered by the deprecated feature versions. This would completely drop support
+ * for the deprecated versions. Such a code change needs to be supplemented by supplying a
+ * suitable higher minVersion value for the feature in the supportedFeatures map.
+ * Example:
+ * - In the example above in step 1, we showed how to deprecate version levels [1, 2] for
+ *   "feature". Now let us assume the following finalized feature in ZK (after the deprecation
+ *   has been carried out):
+ *   {
+ *     "feature" -> FinalizedVersionRange(minVersionLevel=3, maxVersionLevel=5)
+ *   }
+ *   Now, supposing you would like to permanently remove support for feature versions: [1, 2].
+ *   Then, in the supportedFeatures map you should now supply the following:
```

Comment:
Similar here to make `firstActiveVersion = minVersion` as default.

Reply:
As mentioned in above response to a different comment, I've provided an overloaded c'tor now in `org.apache.kafka.common.feature.SupportedVersionRange` that only takes `minVersion` and `maxVersion` as parameters.